semgrep_output_v1.atd

```diff
@@ -2167,6 +2169,37 @@ type resolution_result = [
   | ResolutionError of resolution_error list
 ]
 
+(* ----------------------------- *)
+(* SCA part 4: Symbol analysis *)
+(* ----------------------------- *)
+
+(* "Symbol analysis" is about determining the third-party functions which
+   are used by a given project.
+
+   We may be interested in this for the purposes of analyzing the vulnerability
+   of the repo based on its dependencies, and providing more customized/informed
+   guidance on how to remediate SCA vulns.
+
+   For instance, source code like
+   ```
+   from werkzeug.wrappers import Request
+   ```
+   may be described as using the "symbol" [werkzeug, wrappers, Request].
+ *)
+
+(* A symbol is a FQN. *)
+type symbol = string list
+
+(* We store the location of the usage, because we may want to be able to know how
+   many uses of the symbol there are, and where.
+ *)
+type symbol_usage = {
+  symbol: symbol;
+  loc: location;
```

Comment:
could be a locs: location list; here
to avoid repeating the same symbol again and again (which could save some space).